src/others/xml/lv_xml.c

```diff
@@ -0,0 +1,389 @@
+        XML_ParserFree(parser);
+        return NULL;
+    }
+
+    state.item = state.view;
+
+    if(attrs) {
+        ctx->root_widget->apply_cb(&state, attrs);
+    }
+
+    lv_ll_clear(&state.parent_ll);
+    XML_ParserFree(parser);
+
+    return state.view;
+}
+
+lv_obj_t * lv_xml_create(lv_obj_t * parent, const char * name, const char ** attrs)
+{
+    lv_obj_t * item = NULL;
+
+    /* Select the widget specific parser type based on the name */
+    lv_widget_processor_t * p = lv_xml_widget_get_processor(name);
+    if(p) {
+        lv_xml_parser_state_t state;
+        lv_xml_parser_state_init(&state);
+        state.parent = parent;
+        state.item = p->create_cb(&state, attrs);
+        if(attrs) {
+            p->apply_cb(&state, attrs);
+        }
+        return state.item;
+    }
+
+    lv_xml_component_ctx_t * ctx = lv_xml_component_get_ctx(name);
+    if(ctx) {
+        item = lv_xml_create_from_ctx(parent, NULL, ctx, attrs);
+        return item;
+    }
+
+    /* If it isn't a component either then it is unknown */
+    LV_LOG_WARN("'%s' in not a known widget, element, or component", name);
+    return NULL;
+}
+
+
+lv_result_t lv_xml_register_font(const char * name, const lv_font_t * font)
+{
+    lv_xml_font_t * f = lv_ll_ins_head(&font_ll);
+    f->name = lv_strdup(name);
+    f->font = font;
+
+    return LV_RESULT_OK;
+}
+
+const lv_font_t * lv_xml_get_font(const char * name)
+{
+    lv_xml_font_t * f;
+    LV_LL_READ(&font_ll, f) {
+        if(lv_streq(f->name, name)) return f->font;
+    }
+
+    return NULL;
+}
+
+lv_result_t lv_xml_register_image(const char * name, const void * src)
+{
+    lv_xml_image_t * img = lv_ll_ins_head(&image_ll);
+    img->name = lv_strdup(name);
+    if(lv_image_src_get_type(src) == LV_IMAGE_SRC_FILE) {
+        img->src = lv_strdup(src);
+    }
+    else {
+        img->src = src;
+    }
+
+    return LV_RESULT_OK;
+}
+
+const void * lv_xml_get_image(const char * name)
+{
+    lv_xml_image_t * img;
+    LV_LL_READ(&image_ll, img) {
+        if(lv_streq(img->name, name)) return img->src;
+    }
+
+    return NULL;
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static const char * get_param_type(lv_xml_component_ctx_t * ctx, const char * name)
+{
+    lv_xml_param_t * p;
+    LV_LL_READ(&ctx->param_ll, p) {
+        if(lv_streq(p->name, name)) return p->type;
+    }
+    return NULL;
+}
+
+static const char * get_param_default(lv_xml_component_ctx_t * ctx, const char * name)
+{
+    lv_xml_param_t * p;
+    LV_LL_READ(&ctx->param_ll, p) {
+        if(lv_streq(p->name, name)) return p->def;
+    }
+    return NULL;
+}
+
+static void resolve_params(lv_xml_component_ctx_t * item_ctx, lv_xml_component_ctx_t * parent_ctx,
+                           const char ** item_attrs, const char ** parent_attrs)
+{
+    uint32_t i;
+    for(i = 0; item_attrs[i]; i += 2) {
+        const char * name = item_attrs[i];
+        const char * value = item_attrs[i + 1];
```

Comment:
We can avoid the double pointer by define 

```
struct _lv_xml_attr {
    const char * name;
    const char * value;
}
```

Reply:
Good idea. I will add in a new PR because I don't want to introduce such a large refactoring. (We will release v0.1 this week).